fetch-publication/src/test/java/no/unit/nva/publication/fetch/FetchPublicationHandlerTest.java

```diff
@@ -75,52 +78,58 @@ public void setUp() {
     @DisplayName("handler Returns Ok Response On Valid Input")
     public void handlerReturnsOkResponseOnValidInput() throws IOException, ApiGatewayException {
         Publication publication = createPublication();
-        when(publicationService.getPublication(any(UUID.class)))
-            .thenReturn(publication);
+        when(publicationService.getPublication(any(UUID.class))).thenReturn(publication);
 
-        fetchPublicationHandler.handleRequest(inputStream(), output, context);
+        fetchPublicationHandler.handleRequest(generateHandlerRequest(), output, context);
 
-        GatewayResponse gatewayResponse = objectMapper.readValue(output.toString(), GatewayResponse.class);
+        GatewayResponse<PublicationResponse> gatewayResponse = objectMapper.readValue(output.toString(),
+                PARAMETERIZED_GATEWAY_RESPONSE_TYPE);
         assertEquals(SC_OK, gatewayResponse.getStatusCode());
-        assertTrue(gatewayResponse.getHeaders().keySet().contains(CONTENT_TYPE));
-        assertTrue(gatewayResponse.getHeaders().keySet().contains(ACCESS_CONTROL_ALLOW_ORIGIN));
+        assertTrue(gatewayResponse.getHeaders().containsKey(CONTENT_TYPE));
+        assertTrue(gatewayResponse.getHeaders().containsKey(ACCESS_CONTROL_ALLOW_ORIGIN));
     }
 
     @Test
     @DisplayName("handler Returns NotFound Response On Publication Missing")
     public void handlerReturnsNotFoundResponseOnPublicationMissing() throws IOException, ApiGatewayException {
-        when(publicationService.getPublication(any(UUID.class)))
-            .thenThrow(new NotFoundException("Error"));
+        when(publicationService.getPublication(any(UUID.class))).thenThrow(new NotFoundException("Error"));
 
-        fetchPublicationHandler.handleRequest(inputStream(), output, context);
+        fetchPublicationHandler.handleRequest(generateHandlerRequest(), output, context);
 
-        GatewayResponse gatewayResponse = objectMapper.readValue(output.toString(), GatewayResponse.class);
+        GatewayResponse<PublicationResponse> gatewayResponse = objectMapper.readValue(output.toString(),
+                PARAMETERIZED_GATEWAY_RESPONSE_TYPE);
         assertEquals(SC_NOT_FOUND, gatewayResponse.getStatusCode());
-        assertTrue(gatewayResponse.getHeaders().keySet().contains(CONTENT_TYPE));
-        assertTrue(gatewayResponse.getHeaders().keySet().contains(ACCESS_CONTROL_ALLOW_ORIGIN));
+        assertTrue(gatewayResponse.getHeaders().containsKey(CONTENT_TYPE));
```

Comment:
Usage of `assertThat(gatewayResponse.getHeaders(), hasKey("MyKey")) ` will give a more descriptive error message 
```
Expected: map containing ["MyKey"->ANYTHING]
     but: map was [<Content-Type=application/json>, <Access-Control-Allow-Origin=*>]
```